pkg/controllers/termination/suite_test.go

```diff
@@ -160,6 +161,23 @@ var _ = Describe("Termination", func() {
 			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(node))
 			ExpectNotFound(ctx, env.Client, node)
 		})
+		It("should delete nodes that have a do-not-evict pod that's ignored", func() {
+			pod := test.Pod(test.PodOptions{
+				NodeName:   node.Name,
+				ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{v1alpha5.DoNotEvictPodAnnotationKey: "true"}},
+			})
```

Comment:
Consider testing static pods and a pod that tolerates `NoSchedule`, should just be two more pods creation lines, and adding to the respective lines below.